plugins/zynaddsubfx/zynaddsubfx/src/Misc/QtXmlWrapper.cpp

```diff
@@ -313,7 +331,16 @@ int QtXmlWrapper::loadXMLfile(const std::string &filename)
 char *QtXmlWrapper::doloadfile(const std::string &filename) const
 {
     char  *xmldata = NULL;
-    gzFile gzfile  = gzopen(filename.c_str(), "rb");
+    QFile f(QString::fromUtf8(filename.data(), filename.size()));
+    f.open(QIODevice::ReadOnly);
+#ifdef LMMS_BUILD_WIN32
+    int fd = _dup(f.handle());
+#else
+    int fd = dup(f.handle());
+#endif
+    f.close();
```

Comment:
I wonder if you could offload this logic to a helper function,
```
static int fdopenFromQString(QString filename, QIODevice::OpenMode mode)
```
to avoid duplicating the body in multiple functins, and then leave a comment near that function explaining why the function is necessary (to most readers, this code will look silly and they may be tempted to simplify it back to the way it was before).

Reply:
Good suggestion!